airqo-platform/AirQo-api
# Fix device monitoring cronjobs

**What does this PR do?**
  - This PR addresses the issues that have been preventing cronjobs for calculating network uptime and offline /online devices status to run successfully at their scheduled time slots.
-  Also applies a fix to endpoint for device uptime to consider channelIds as strings

**What JIRA task is related to this PR?**
   N/A
**How do I test out this PR?**
 - git fetch branch fix-device-monitoring-cronjobs
 - change directory to device-monitoring (device monitoring service)
 - run pip install -r requirements.txt to get all the relevant packages.
 - Afterwards run flask run to start the backend service
 - Use any of the rest-api client to test out the endpoint
 -  then run the  latest locate/netmanager product frontend (from staging) and view the graphs for network uptime/ offline/online status and check for the last updated date.. for offline/online device it should be last hour and for network uptime the beginning of day 
- Confirm that when device details page is opened, device uptime graph displays 
- to view device uptime, device-restry backend service needs to also be running in parallel with this backend service.
**Which end points should are ready for testing?**
   - /api/v1/monitor/device/uptime/718029
